Sources/Hyperspace/Extensions/URL+Additions.swift

```diff
@@ -17,3 +18,37 @@ extension URL {
         return url
     }
 }
+
+public extension URL {
```

Comment:
Now that I'm seeing this as an extension, I'm wondering if we should maybe take a hint from `Foundation` and `JSONEncoder` and create a `URLQueryEncoder` (`URLQueryParameterEncoder`?). I'm sitting here looking at `JSONEncoder` for another project and was seeing the parallels (minus needing decoding support). What are your thoughts?

Reply:
Now that you mention it, it does make sense. It's a simplified use case of an encoder, but an encoder. My only worry in this case is it doesn't make as much sense to conform to `Encoder`. We could make a `URLQueryParameterEncoder` that is an encoder in name only?

Reply:
Yeah, `Encoder` does seem a bit heavy for what we need. I'm fine with naming it that way, but not actually investing in making it conform to `Encoder`.

Reply:
Made the change. Left a category on `URL` which makes it easy to use the `URLQueryParameterEncoder` to append the query.